Accessibility. Crash on Windows when traversing elements

Compose 1.6.10-rc01, Windows 11, NVDA 2023.2

1. run run1
2. traverse with hot keys
3. see crash:
```
Caused by: java.lang.IndexOutOfBoundsException: index: -3, size: 1
	at kotlin.collections.AbstractList$Companion.checkElementIndex$kotlin_stdlib(AbstractList.kt:108)
	at kotlin.collections.builders.ListBuilder.get(ListBuilder.kt:59)
	at androidx.compose.ui.platform.a11y.ComposeAccessible$ComposeAccessibleComponent.getAccessibleChild(ComposeAccessible.kt:337)
	at androidx.compose.ui.platform.a11y.ComposeAccessible$ComposeAccessibleComponent.getAccessibleAt(ComposeAccessible.kt:380)
	at androidx.compose.ui.platform.a11y.ComposeAccessible$ComposeAccessibleComponent.getAccessibleAt(ComposeAccessible.kt:381)
	at androidx.compose.ui.platform.a11y.ComposeAccessible$ComposeAccessibleComponent.getAccessibleAt(ComposeAccessible.kt:381)
	at androidx.compose.ui.platform.a11y.ComposeSceneAccessible$ComposeSceneAccessibleContext.getAccessibleAt(ComposeSceneAccessible.kt:89)
	at jdk.accessibility/com.sun.java.accessibility.util.EventQueueMonitor.getAccessibleAt(EventQueueMonitor.java:441)
	at jdk.accessibility/com.sun.java.accessibility.internal.AccessBridge$12.call(AccessBridge.java:581)
	at jdk.accessibility/com.sun.java.accessibility.internal.AccessBridge$12.call(AccessBridge.java:578)
	at jdk.accessibility/com.sun.java.accessibility.internal.AccessBridge$InvocationUtils$CallableWrapper.run(AccessBridge.java:7334)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:308)
```

--- a/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/platform/a11y/ComposeAccessible.kt
+++ b/compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/platform/a11y/ComposeAccessible.kt
@@ -332,9 +332,13 @@ internal class ComposeAccessible(
         }
 
         override fun getAccessibleChild(i: Int): Accessible? {
-            return semanticsNode.replacedChildren.getOrNull(i)?.id?.let { id ->
-                controller.accessibleByNodeId(id)
-            } ?: auxiliaryChildren[i - semanticsNode.replacedChildren.size]
+            val replacedChildren = semanticsNode.replacedChildren
+            val replacedChildrenSize = replacedChildren.size
+            return if (i < replacedChildrenSize) {
+                controller.accessibleByNodeId(replacedChildren[i].id)
+            } else {
+                auxiliaryChildren[i - replacedChildrenSize]
+            }
         }
 
         override fun getLocale(): Locale = Locale.getDefault()

